components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/factory/RegistrationRequestFactoryTest.java

```diff
@@ -0,0 +1,289 @@
+                        "expected response type is not found in registrationRequestProfile");
+            }
+        }
+        assertEquals(registrationRequestProfile.getOwner(), ownerName,
+                "expected owner name is not found in registrationRequestProfile");
+        PrivilegedCarbonContext.endTenantFlow();
+    }
+
+    @Test(expectedExceptions = IdentityException.class)
+    public void testCreateWithEmptyRedirectUri() throws Exception {
+
+        String grantType = "implicit";
+        int redirectUrl = 0;
+        JSONObject jsonObject = new JSONObject();
+
+        jsonObject.put(RegistrationRequest.RegisterRequestConstant.GRANT_TYPES, grantType);
+        jsonObject.put(RegistrationRequest.RegisterRequestConstant.REDIRECT_URIS, redirectUrl);
+
+        RegistrationRequestProfile registrationRequestProfile = new RegistrationRequestProfile();
+
+        whenNew(RegistrationRequestProfile.class).withNoArguments().thenReturn(registrationRequestProfile);
+
+        suppress(methodsDeclaredIn(HttpIdentityRequestFactory.class));
+
+        when(mockHttpRequest.getReader()).thenReturn(mockReader);
+        whenNew(JSONParser.class).withNoArguments().thenReturn(jsonParser);
+
+        when(jsonParser.parse(mockReader)).thenReturn(jsonObject);
+
+        startTenantFlow();
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setUsername(ownerName);
+
+        registrationRequestFactory.create(mockRegistrationRequestBuilder, mockHttpRequest, mockHttpResponse);
+        PrivilegedCarbonContext.endTenantFlow();
+    }
+
+    @Test
+    public void testHandleException() throws Exception {
+
+        whenNew(HttpIdentityResponse.HttpIdentityResponseBuilder.class).withNoArguments().thenReturn
+                (mockHttpIdentityResponseBuilder);
+
+        final Integer[] statusCode = new Integer[1];
+        doAnswer(new Answer<Object>() {
+            @Override
+            public Object answer(InvocationOnMock invocation) throws Throwable {
+
+                statusCode[0] = (Integer) invocation.getArguments()[0];
+                return null;
+            }
+        }).when(mockHttpIdentityResponseBuilder).setStatusCode(anyInt());
+
+        final String[] header = new String[1];
+        doAnswer(new Answer<Object>() {
+            @Override
+            public Object answer(InvocationOnMock invocation) throws Throwable {
+
+                header[0] = (String) invocation.getArguments()[1];
+                return null;
+            }
+        }).when(mockHttpIdentityResponseBuilder).addHeader(anyString(), anyString());
+
+        FrameworkClientException exception = mock(FrameworkClientException.class);
+        when(exception.getMessage()).thenReturn(dummyDescription);
+        registrationRequestFactory.handleException(exception, mockHttpRequest, mockHttpResponse);
+
+        assertEquals(header[0], MediaType.APPLICATION_JSON, "Header doesn't match with " +
```

Comment:
Can you guarantee that the first header will be this ? I had experience that things come in different order.